cmd/olm/cleanup.go

```diff
@@ -239,3 +256,31 @@ func crossNamespaceOwnerReferenceRemoval(kind string, uidNamespaces map[types.UI
 		return
 	}
 }
+
+// checkAPIServiceLabels checks the labels of existing APIService objects to ensure ownership
+// solves a potential contention issues for packageservers (which own the same APIService) during upgrades
```

Comment:
If you wish to include a note highlighting the problems fixed by the function, could it be more robust?

I would personally suggest just writing what the function does.